Examples/Cocoapods/SpeechToText/Speech/AudioController.swift

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+import AVFoundation
+
+protocol AudioControllerDelegate {
+  func processSampleData(_ data: Data) -> Void
+}
+
+class AudioController {
+  var remoteIOUnit: AudioComponentInstance? // optional to allow it to be an inout argument
+  var delegate: AudioControllerDelegate!
+  
+  static var sharedInstance = AudioController()
+  
+  deinit {
+    AudioComponentInstanceDispose(remoteIOUnit!)
+  }
+  
+  func prepare(specifiedSampleRate: Int) -> OSStatus {
+    var status = noErr
+    
+    let session = AVAudioSession.sharedInstance()
+    do {
+      try session.setCategory(AVAudioSession.Category.record)
+      try session.setPreferredIOBufferDuration(10)
+    } catch {
+      return -1
+    }
+    
+    var sampleRate = session.sampleRate
+    print("hardware sample rate = \(sampleRate), using specified rate = \(specifiedSampleRate)")
+    sampleRate = Double(specifiedSampleRate)
+    
+    // Describe the RemoteIO unit
+    var audioComponentDescription = AudioComponentDescription()
+    audioComponentDescription.componentType = kAudioUnitType_Output
+    audioComponentDescription.componentSubType = kAudioUnitSubType_RemoteIO
+    audioComponentDescription.componentManufacturer = kAudioUnitManufacturer_Apple
+    audioComponentDescription.componentFlags = 0
+    audioComponentDescription.componentFlagsMask = 0
+    
+    // Get the RemoteIO unit
+    let remoteIOComponent = AudioComponentFindNext(nil, &audioComponentDescription)
+    status = AudioComponentInstanceNew(remoteIOComponent!, &remoteIOUnit)
+    if (status != noErr) {
+      return status
+    }
+    
+    let bus1: AudioUnitElement = 1
+    var oneFlag: UInt32 = 1
+    
+    // Configure the RemoteIO unit for input
+    status = AudioUnitSetProperty(remoteIOUnit!,
+                                  kAudioOutputUnitProperty_EnableIO,
+                                  kAudioUnitScope_Input,
+                                  bus1,
+                                  &oneFlag,
+                                  UInt32(MemoryLayout<UInt32>.size))
+    if (status != noErr) {
+      return status
+    }
+    
+    // Set format for mic input (bus 1) on RemoteIO's output scope
+    var asbd = AudioStreamBasicDescription()
+    asbd.mSampleRate = sampleRate
+    asbd.mFormatID = kAudioFormatLinearPCM
```

Comment:
Worth adding a comment saying that the initial request config is linked to this config.